Flaky test ModelingAssessmentIntegrationTest > testAutomaticAssessmentActivityDiagramUponAssessmentSubmission

### Describe the bug

The test ModelingAssessmentIntegrationTest > testAutomaticAssessmentActivityDiagramUponAssessmentSubmission is flaky on Github actions

### To Reproduce

Run the server tests in any PR

### Expected behavior

The test should pass reliably

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Other (specify in "Additional context")

### Additional context

_No response_

### Relevant log output

```shell
2021-07-15T20:55:46.2901330Z ModelingAssessmentIntegrationTest > testAutomaticAssessmentActivityDiagramUponAssessmentSubmission() STANDARD_OUT
2021-07-15T20:55:46.2905006Z     2021-07-15 20:55:46.193  INFO 1935 --- [    Test worker] d.t.i.w.a.s.compass.CompassService       : ModelCluster: start building clusters of 2 submissions for modeling exercise 3033
2021-07-15T20:55:46.2907633Z     2021-07-15 20:55:46.206  INFO 1935 --- [    Test worker] d.t.i.w.a.s.compass.CompassService       : ModelClusterTimelog: building clusters of 2 submissions for modeling exercise 3033 done in 15.71ms
2021-07-15T20:55:46.2910464Z     2021-07-15 20:55:46.233  INFO 1935 --- [    Test worker] d.t.i.w.a.s.compass.CompassService       : ModelClusterTimelog: building and saving clusters of 2 submissions for exercise 3033 done in 43.12ms
2021-07-15T20:55:46.2913773Z     2021-07-15 20:55:46.234 DEBUG 1935 --- [    Test worker] d.t.i.w.artemis.config.ApiVersionFilter  : Adding Version to Request PUT /api/modeling-submissions/3579/result/0/assessment
2021-07-15T20:55:46.3903084Z     2021-07-15 20:55:46.324 ERROR 1935 --- [    Test worker] o.z.problem.spring.common.AdviceTraits   : Internal Server Error
2021-07-15T20:55:46.3915321Z 
2021-07-15T20:55:46.3925103Z     org.springframework.dao.InvalidDataAccessApiUsageException: Multiple representations of the same entity [de.tum.in.www1.artemis.domain.Feedback#2386] are being merged. Detached: [Feedback{id=2386, text='null', detailText='null', reference='ActivityActionNode:1c0576f6-89b0-4b13-96d9-e1a0a227e8e1', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]; Detached: [Feedback{id=2386, text='correct activity, but missing elements', detailText='null', reference='Activity:cfcb7b17-0500-4654-89bb-a7082d5c3b6d', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]; nested exception is java.lang.IllegalStateException: Multiple representations of the same entity [de.tum.in.www1.artemis.domain.Feedback#2386] are being merged. Detached: [Feedback{id=2386, text='null', detailText='null', reference='ActivityActionNode:1c0576f6-89b0-4b13-96d9-e1a0a227e8e1', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]; Detached: [Feedback{id=2386, text='correct activity, but missing elements', detailText='null', reference='Activity:cfcb7b17-0500-4654-89bb-a7082d5c3b6d', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]
2021-07-15T20:55:46.3942721Z     	at org.springframework.orm.jpa.EntityManagerFactoryUtils.convertJpaAccessExceptionIfPossible(EntityManagerFactoryUtils.java:371)
2021-07-15T20:55:46.3947734Z     	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:235)
2021-07-15T20:55:46.3952594Z     	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:551)
2021-07-15T20:55:46.3958566Z     	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61)
2021-07-15T20:55:46.3963557Z     	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242)
2021-07-15T20:55:46.3967705Z     	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:152)
2021-07-15T20:55:46.3972064Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.3976857Z     	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:174)
2021-07-15T20:55:46.3981647Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.3985591Z     	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
2021-07-15T20:55:46.3988892Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.3992073Z     	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215)
2021-07-15T20:55:46.3993820Z     	at jdk.proxy3/jdk.proxy3.$Proxy312.save(Unknown Source)
2021-07-15T20:55:46.3995791Z     	at de.tum.in.www1.artemis.service.AssessmentService.saveManualAssessment(AssessmentService.java:298)
2021-07-15T20:55:46.3998387Z     	at de.tum.in.www1.artemis.service.AssessmentService$$FastClassBySpringCGLIB$$84002fe7.invoke(<generated>)
2021-07-15T20:55:46.4000454Z     	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
2021-07-15T20:55:46.4082205Z     	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688)
2021-07-15T20:55:46.4084850Z     	at de.tum.in.www1.artemis.service.AssessmentService$$EnhancerBySpringCGLIB$$85f9d60c.saveManualAssessment(<generated>)
2021-07-15T20:55:46.4087502Z     	at de.tum.in.www1.artemis.web.rest.AssessmentResource.saveAssessment(AssessmentResource.java:114)
2021-07-15T20:55:46.4091015Z     	at de.tum.in.www1.artemis.web.rest.ModelingAssessmentResource.saveModelingAssessment(ModelingAssessmentResource.java:99)
2021-07-15T20:55:46.4094420Z     	at de.tum.in.www1.artemis.web.rest.ModelingAssessmentResource$$FastClassBySpringCGLIB$$e75cf5f6.invoke(<generated>)
2021-07-15T20:55:46.4096737Z     	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
2021-07-15T20:55:46.4099119Z     	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779)
2021-07-15T20:55:46.4102243Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
2021-07-15T20:55:46.4105443Z     	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
2021-07-15T20:55:46.4111700Z     	at org.springframework.security.access.intercept.aopalliance.MethodSecurityInterceptor.invoke(MethodSecurityInterceptor.java:61)
2021-07-15T20:55:46.4116081Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.4119044Z     	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
2021-07-15T20:55:46.4121750Z     	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)
2021-07-15T20:55:46.4124579Z     	at de.tum.in.www1.artemis.web.rest.ModelingAssessmentResource$$EnhancerBySpringCGLIB$$f5f918c0.saveModelingAssessment(<generated>)
2021-07-15T20:55:46.4127892Z     	at jdk.internal.reflect.GeneratedMethodAccessor2285.invoke(Unknown Source)
2021-07-15T20:55:46.4130487Z     	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-07-15T20:55:46.4132538Z     	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
2021-07-15T20:55:46.4134871Z     	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:197)
2021-07-15T20:55:46.4138688Z     	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:141)
2021-07-15T20:55:46.4143749Z     	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:106)
2021-07-15T20:55:46.4150012Z     	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:894)
2021-07-15T20:55:46.4156114Z     	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
2021-07-15T20:55:46.4241539Z     	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
2021-07-15T20:55:46.4245149Z     	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1063)
2021-07-15T20:55:46.4247784Z     	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
2021-07-15T20:55:46.4250591Z     	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
2021-07-15T20:55:46.4253228Z     	at org.springframework.web.servlet.FrameworkServlet.doPut(FrameworkServlet.java:920)
2021-07-15T20:55:46.4255115Z     	at javax.servlet.http.HttpServlet.service(HttpServlet.java:526)
2021-07-15T20:55:46.4257054Z     	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
2021-07-15T20:55:46.4259849Z     	at org.springframework.test.web.servlet.TestDispatcherServlet.service(TestDispatcherServlet.java:72)
2021-07-15T20:55:46.4262158Z     	at javax.servlet.http.HttpServlet.service(HttpServlet.java:590)
2021-07-15T20:55:46.4264016Z     	at org.springframework.mock.web.MockFilterChain$ServletFilterProxy.doFilter(MockFilterChain.java:167)
2021-07-15T20:55:46.4266282Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4268808Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:113)
2021-07-15T20:55:46.4271307Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4273536Z     	at de.tum.in.www1.artemis.config.ApiVersionFilter.doFilter(ApiVersionFilter.java:46)
2021-07-15T20:55:46.4275757Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4278820Z     	at org.springframework.web.servlet.resource.ResourceUrlEncodingFilter.doFilter(ResourceUrlEncodingFilter.java:67)
2021-07-15T20:55:46.4282235Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4284201Z     	at io.sentry.spring.SentryUserFilter.doFilter(SentryUserFilter.java:56)
2021-07-15T20:55:46.4286180Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4288577Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:327)
2021-07-15T20:55:46.4292102Z     	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:115)
2021-07-15T20:55:46.4296626Z     	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:81)
2021-07-15T20:55:46.4300293Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4303551Z     	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:121)
2021-07-15T20:55:46.4307526Z     	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:115)
2021-07-15T20:55:46.4312574Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4315800Z     	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:126)
2021-07-15T20:55:46.4319521Z     	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:81)
2021-07-15T20:55:46.4403049Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4406879Z     	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:105)
2021-07-15T20:55:46.4410600Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4415974Z     	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:149)
2021-07-15T20:55:46.4420311Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4423691Z     	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
2021-07-15T20:55:46.4427039Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4429317Z     	at de.tum.in.www1.artemis.security.jwt.JWTFilter.doFilter(JWTFilter.java:36)
2021-07-15T20:55:46.4431598Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4434106Z     	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91)
2021-07-15T20:55:46.4436684Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
2021-07-15T20:55:46.4439387Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4442595Z     	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:103)
2021-07-15T20:55:46.4446177Z     	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:89)
2021-07-15T20:55:46.4449239Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4452330Z     	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
2021-07-15T20:55:46.4456030Z     	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
2021-07-15T20:55:46.4459326Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
2021-07-15T20:55:46.4462272Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4465948Z     	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:110)
2021-07-15T20:55:46.4470735Z     	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:80)
2021-07-15T20:55:46.4474403Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4564388Z     	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:55)
2021-07-15T20:55:46.4569481Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
2021-07-15T20:55:46.4572193Z     	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
2021-07-15T20:55:46.4575041Z     	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:211)
2021-07-15T20:55:46.4577835Z     	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:183)
2021-07-15T20:55:46.4581074Z     	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
2021-07-15T20:55:46.4590797Z     	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
2021-07-15T20:55:46.4593486Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4596270Z     	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
2021-07-15T20:55:46.4599368Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
2021-07-15T20:55:46.4602624Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4605245Z     	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
2021-07-15T20:55:46.4608146Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
2021-07-15T20:55:46.4610674Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4614214Z     	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)
2021-07-15T20:55:46.4618550Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
2021-07-15T20:55:46.4621089Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4624008Z     	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
2021-07-15T20:55:46.4628074Z     	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
2021-07-15T20:55:46.4630617Z     	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
2021-07-15T20:55:46.4632707Z     	at org.springframework.test.web.servlet.MockMvc.perform(MockMvc.java:183)
2021-07-15T20:55:46.4634720Z     	at de.tum.in.www1.artemis.util.RequestUtilService.put(RequestUtilService.java:333)
2021-07-15T20:55:46.4637747Z     	at de.tum.in.www1.artemis.ModelingAssessmentIntegrationTest.createAssessment(ModelingAssessmentIntegrationTest.java:1389)
2021-07-15T20:55:46.4697640Z     	at de.tum.in.www1.artemis.ModelingAssessmentIntegrationTest.testAutomaticAssessmentActivityDiagramUponAssessmentSubmission(ModelingAssessmentIntegrationTest.java:686)
2021-07-15T20:55:46.4702870Z     	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-07-15T20:55:46.4705221Z     	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
2021-07-15T20:55:46.4708226Z     	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-07-15T20:55:46.4710570Z     	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
2021-07-15T20:55:46.4712931Z     	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
2021-07-15T20:55:46.4720184Z     	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
2021-07-15T20:55:46.4724375Z     	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
2021-07-15T20:55:46.4727489Z     	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
2021-07-15T20:55:46.4803470Z     	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
2021-07-15T20:55:46.4806994Z     	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
2021-07-15T20:55:46.4810106Z     	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
2021-07-15T20:55:46.4812721Z     	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
2021-07-15T20:55:46.4815571Z     	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
2021-07-15T20:55:46.4818848Z     	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
2021-07-15T20:55:46.4823467Z     	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
2021-07-15T20:55:46.4827224Z     	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
2021-07-15T20:55:46.4830217Z     	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
2021-07-15T20:55:46.4832803Z     	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
2021-07-15T20:55:46.4835890Z     	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
2021-07-15T20:55:46.4839415Z     	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2021-07-15T20:55:46.4843310Z     	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
2021-07-15T20:55:46.4847037Z     	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
2021-07-15T20:55:46.4850453Z     	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
2021-07-15T20:55:46.4853720Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2021-07-15T20:55:46.4857022Z     	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2021-07-15T20:55:46.4860332Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
2021-07-15T20:55:46.4862989Z     	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2021-07-15T20:55:46.4865664Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
2021-07-15T20:55:46.4868951Z     	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2021-07-15T20:55:46.4872541Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
2021-07-15T20:55:46.4876323Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
2021-07-15T20:55:46.4878518Z     	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2021-07-15T20:55:46.4918369Z     	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2021-07-15T20:55:46.4933554Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
2021-07-15T20:55:46.4936968Z     	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2021-07-15T20:55:46.4940291Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
2021-07-15T20:55:46.4942967Z     	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2021-07-15T20:55:46.4945846Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
2021-07-15T20:55:46.4949140Z     	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2021-07-15T20:55:46.4952750Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
2021-07-15T20:55:46.4956066Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
2021-07-15T20:55:46.4958043Z     	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2021-07-15T20:55:46.4962057Z     	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2021-07-15T20:55:46.4966883Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
2021-07-15T20:55:46.4970223Z     	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2021-07-15T20:55:46.4973500Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
2021-07-15T20:55:46.4976154Z     	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2021-07-15T20:55:46.4978821Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
2021-07-15T20:55:46.4982116Z     	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2021-07-15T20:55:46.4985721Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
2021-07-15T20:55:46.4989060Z     	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
2021-07-15T20:55:46.4993698Z     	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2021-07-15T20:55:46.4998553Z     	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2021-07-15T20:55:46.5082968Z     	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2021-07-15T20:55:46.5086744Z     	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
2021-07-15T20:55:46.5090197Z     	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
2021-07-15T20:55:46.5093653Z     	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
2021-07-15T20:55:46.5097723Z     	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
2021-07-15T20:55:46.5102155Z     	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
2021-07-15T20:55:46.5105054Z     	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
2021-07-15T20:55:46.5107594Z     	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
2021-07-15T20:55:46.5111879Z     	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
2021-07-15T20:55:46.5117714Z     	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
2021-07-15T20:55:46.5123141Z     	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
2021-07-15T20:55:46.5128623Z     	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
2021-07-15T20:55:46.5131457Z     	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-07-15T20:55:46.5133818Z     	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
2021-07-15T20:55:46.5136547Z     	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-07-15T20:55:46.5138603Z     	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
2021-07-15T20:55:46.5140428Z     	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
2021-07-15T20:55:46.5142873Z     	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
2021-07-15T20:55:46.5145722Z     	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
2021-07-15T20:55:46.5148833Z     	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
2021-07-15T20:55:46.5150728Z     	at jdk.proxy2/jdk.proxy2.$Proxy5.stop(Unknown Source)
2021-07-15T20:55:46.5152360Z     	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:135)
2021-07-15T20:55:46.5154628Z     	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-07-15T20:55:46.5156976Z     	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
2021-07-15T20:55:46.5159713Z     	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-07-15T20:55:46.5229956Z     	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
2021-07-15T20:55:46.5231808Z     	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
2021-07-15T20:55:46.5234243Z     	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
2021-07-15T20:55:46.5237646Z     	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
2021-07-15T20:55:46.5242116Z     	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
2021-07-15T20:55:46.5245300Z     	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
2021-07-15T20:55:46.5247563Z     	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2021-07-15T20:55:46.5249890Z     	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
2021-07-15T20:55:46.5252096Z     	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
2021-07-15T20:55:46.5254580Z     	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
2021-07-15T20:55:46.5256745Z     	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
2021-07-15T20:55:46.5258388Z     	at java.base/java.lang.Thread.run(Thread.java:831)
2021-07-15T20:55:46.5292163Z     Caused by: java.lang.IllegalStateException: Multiple representations of the same entity [de.tum.in.www1.artemis.domain.Feedback#2386] are being merged. Detached: [Feedback{id=2386, text='null', detailText='null', reference='ActivityActionNode:1c0576f6-89b0-4b13-96d9-e1a0a227e8e1', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]; Detached: [Feedback{id=2386, text='correct activity, but missing elements', detailText='null', reference='Activity:cfcb7b17-0500-4654-89bb-a7082d5c3b6d', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]
2021-07-15T20:55:46.5312738Z     	at org.hibernate.event.internal.EntityCopyNotAllowedObserver.entityCopyDetected(EntityCopyNotAllowedObserver.java:45)
2021-07-15T20:55:46.5315826Z     	at org.hibernate.event.internal.MergeContext.put(MergeContext.java:245)
2021-07-15T20:55:46.5318790Z     	at org.hibernate.event.internal.DefaultMergeEventListener.entityIsDetached(DefaultMergeEventListener.java:322)
2021-07-15T20:55:46.5322273Z     	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:172)
2021-07-15T20:55:46.5326091Z     	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:118)
2021-07-15T20:55:46.5330072Z     	at org.hibernate.internal.SessionImpl.fireMerge(SessionImpl.java:842)
2021-07-15T20:55:46.5331721Z     	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:815)
2021-07-15T20:55:46.5333354Z     	at org.hibernate.engine.spi.CascadingActions$6.cascade(CascadingActions.java:261)
2021-07-15T20:55:46.5335180Z     	at org.hibernate.engine.internal.Cascade.cascadeToOne(Cascade.java:510)
2021-07-15T20:55:46.5337216Z     	at org.hibernate.engine.internal.Cascade.cascadeAssociation(Cascade.java:434)
2021-07-15T20:55:46.5339319Z     	at org.hibernate.engine.internal.Cascade.cascadeProperty(Cascade.java:220)
2021-07-15T20:55:46.5341654Z     	at org.hibernate.engine.internal.Cascade.cascadeCollectionElements(Cascade.java:543)
2021-07-15T20:55:46.5344030Z     	at org.hibernate.engine.internal.Cascade.cascadeCollection(Cascade.java:474)
2021-07-15T20:55:46.5346188Z     	at org.hibernate.engine.internal.Cascade.cascadeAssociation(Cascade.java:437)
2021-07-15T20:55:46.5348299Z     	at org.hibernate.engine.internal.Cascade.cascadeProperty(Cascade.java:220)
2021-07-15T20:55:46.5350126Z     	at org.hibernate.engine.internal.Cascade.cascade(Cascade.java:153)
2021-07-15T20:55:46.5352724Z     	at org.hibernate.event.internal.DefaultMergeEventListener.cascadeOnMerge(DefaultMergeEventListener.java:518)
2021-07-15T20:55:46.5420343Z     	at org.hibernate.event.internal.DefaultMergeEventListener.entityIsDetached(DefaultMergeEventListener.java:346)
2021-07-15T20:55:46.5423774Z     	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:172)
2021-07-15T20:55:46.5426902Z     	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:70)
2021-07-15T20:55:46.5431428Z     	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107)
2021-07-15T20:55:46.5435034Z     	at org.hibernate.internal.SessionImpl.fireMerge(SessionImpl.java:822)
2021-07-15T20:55:46.5436656Z     	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:809)
2021-07-15T20:55:46.5438485Z     	at jdk.internal.reflect.GeneratedMethodAccessor271.invoke(Unknown Source)
2021-07-15T20:55:46.5441115Z     	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-07-15T20:55:46.5443153Z     	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
2021-07-15T20:55:46.5483571Z     	at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:311)
2021-07-15T20:55:46.5485933Z     	at jdk.proxy3/jdk.proxy3.$Proxy281.merge(Unknown Source)
2021-07-15T20:55:46.5488391Z     	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.save(SimpleJpaRepository.java:600)
2021-07-15T20:55:46.5491353Z     	at jdk.internal.reflect.GeneratedMethodAccessor282.invoke(Unknown Source)
2021-07-15T20:55:46.5493921Z     	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-07-15T20:55:46.5496021Z     	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
2021-07-15T20:55:46.5498985Z     	at org.springframework.data.repository.core.support.RepositoryMethodInvoker$RepositoryFragmentMethodInvoker.lambda$new$0(RepositoryMethodInvoker.java:289)
2021-07-15T20:55:46.5503458Z     	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.doInvoke(RepositoryMethodInvoker.java:137)
2021-07-15T20:55:46.5507664Z     	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.invoke(RepositoryMethodInvoker.java:121)
2021-07-15T20:55:46.5511776Z     	at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:529)
2021-07-15T20:55:46.5515619Z     	at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:285)
2021-07-15T20:55:46.5520058Z     	at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:599)
2021-07-15T20:55:46.5524428Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.5528756Z     	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.doInvoke(QueryExecutorMethodInterceptor.java:163)
2021-07-15T20:55:46.5542394Z     	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.invoke(QueryExecutorMethodInterceptor.java:138)
2021-07-15T20:55:46.5546686Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.5550897Z     	at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:80)
2021-07-15T20:55:46.5555123Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.5558658Z     	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
2021-07-15T20:55:46.5683408Z     	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
2021-07-15T20:55:46.5687904Z     	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
2021-07-15T20:55:46.5691481Z     	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2021-07-15T20:55:46.5696198Z     	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:137)
2021-07-15T20:55:46.5698854Z     	... 196 common frames omitted
2021-07-15T20:55:46.5699160Z 
2021-07-15T20:55:47.3897093Z 
2021-07-15T20:55:47.3898429Z     MockHttpServletRequest:
2021-07-15T20:55:47.3899239Z           HTTP Method = PUT
2021-07-15T20:55:47.3901065Z           Request URI = /api/modeling-submissions/3579/result/0/assessment
2021-07-15T20:55:47.3901924Z            Parameters = {submit=[true]}
2021-07-15T20:55:47.3902983Z               Headers = [Content-Type:"application/json;charset=UTF-8", Content-Length:"3484"]
2021-07-15T20:55:47.3903704Z                  Body = [ {
2021-07-15T20:55:47.3904081Z       "id" : 2378,
2021-07-15T20:55:47.3904633Z       "text" : "correct activity, but missing elements",
2021-07-15T20:55:47.3905660Z       "reference" : "Activity:cfcb7b17-0500-4654-89bb-a7082d5c3b6d",
2021-07-15T20:55:47.3906343Z       "credits" : 0.0,
2021-07-15T20:55:47.3906786Z       "positive" : true,
2021-07-15T20:55:47.3907245Z       "type" : "MANUAL"
2021-07-15T20:55:47.3907931Z     }, {
2021-07-15T20:55:47.3908289Z       "id" : 2379,
2021-07-15T20:55:47.3909271Z       "reference" : "ActivityControlFlow:fa0285e3-a3e6-4281-8c0e-223e5758106e",
2021-07-15T20:55:47.3910047Z       "credits" : 0.5,
2021-07-15T20:55:47.3910481Z       "positive" : true,
2021-07-15T20:55:47.3911055Z       "type" : "MANUAL"
2021-07-15T20:55:47.3911431Z     }, {
2021-07-15T20:55:47.3911784Z       "id" : 2380,
2021-07-15T20:55:47.3912179Z       "text" : "good",
2021-07-15T20:55:47.3913026Z       "reference" : "Activity:f85b009b-3320-4c73-9c7b-0d4bbe7e74f9",
2021-07-15T20:55:47.3913677Z       "credits" : 0.5,
2021-07-15T20:55:47.3914124Z       "positive" : true,
2021-07-15T20:55:47.3914567Z       "type" : "MANUAL"
2021-07-15T20:55:47.3914957Z     }, {
2021-07-15T20:55:47.3915304Z       "id" : 2381,
2021-07-15T20:55:47.3916344Z       "reference" : "ActivityMergeNode:02c342c2-53d4-4bd7-9e08-774dbaa97ca6",
2021-07-15T20:55:47.3917086Z       "credits" : 0.5,
2021-07-15T20:55:47.3917531Z       "positive" : true,
2021-07-15T20:55:47.3917994Z       "type" : "MANUAL"
2021-07-15T20:55:47.3918377Z     }, {
2021-07-15T20:55:47.3918740Z       "id" : 2382,
2021-07-15T20:55:47.3919683Z       "reference" : "ActivityForkNode:06c289b0-0a7d-4f9b-abef-3027a06330aa",
2021-07-15T20:55:47.3920424Z       "credits" : 0.0,
2021-07-15T20:55:47.3932265Z       "positive" : true,
2021-07-15T20:55:47.3932775Z       "type" : "MANUAL"
2021-07-15T20:55:47.3933158Z     }, {
2021-07-15T20:55:47.3933510Z       "id" : 2383,
2021-07-15T20:55:47.3934758Z       "reference" : "ActivityInitialNode:2d1fbb07-bf49-43c1-a55f-35437e2cfcdc",
2021-07-15T20:55:47.3935652Z       "credits" : 0.0,
2021-07-15T20:55:47.3936052Z       "positive" : true,
2021-07-15T20:55:47.3936477Z       "type" : "MANUAL"
2021-07-15T20:55:47.3936842Z     }, {
2021-07-15T20:55:47.3937172Z       "id" : 2384,
2021-07-15T20:55:47.3937795Z       "text" : "not necessary, but fine",
2021-07-15T20:55:47.3938923Z       "reference" : "ActivityMergeNode:e81a38a1-9ceb-46c9-b9e4-f8d274c76085",
2021-07-15T20:55:47.3939645Z       "credits" : 0.0,
2021-07-15T20:55:47.3940053Z       "positive" : true,
2021-07-15T20:55:47.3940472Z       "type" : "MANUAL"
2021-07-15T20:55:47.3940818Z     }, {
2021-07-15T20:55:47.3941321Z       "id" : 2385,
2021-07-15T20:55:47.3942238Z       "reference" : "ActivityFinalNode:5019a714-3bef-4e4e-b649-b759c2b8778a",
2021-07-15T20:55:47.3942969Z       "credits" : 0.0,
2021-07-15T20:55:47.3943392Z       "positive" : true,
2021-07-15T20:55:47.3943847Z       "type" : "MANUAL"
2021-07-15T20:55:47.3944233Z     }, {
2021-07-15T20:55:47.3944564Z       "id" : 2386,
2021-07-15T20:55:47.3946063Z       "reference" : "ActivityActionNode:1c0576f6-89b0-4b13-96d9-e1a0a227e8e1",
2021-07-15T20:55:47.3946814Z       "credits" : 0.5,
2021-07-15T20:55:47.3947243Z       "positive" : true,
2021-07-15T20:55:47.3947706Z       "type" : "MANUAL"
2021-07-15T20:55:47.3948096Z     }, {
2021-07-15T20:55:47.3948446Z       "id" : 2387,
2021-07-15T20:55:47.3948840Z       "text" : "wrong name",
2021-07-15T20:55:47.3949852Z       "reference" : "ActivityActionNode:056d0fe3-942a-4719-9b59-4c52443adcd5",
2021-07-15T20:55:47.3950712Z       "credits" : -0.5,
2021-07-15T20:55:47.3951149Z       "positive" : false,
2021-07-15T20:55:47.3951603Z       "type" : "MANUAL"
2021-07-15T20:55:47.3951989Z     }, {
2021-07-15T20:55:47.3952320Z       "id" : 2389,
2021-07-15T20:55:47.3952726Z       "text" : "awesome",
2021-07-15T20:55:47.3953658Z       "reference" : "ActivityObjectNode:525b1697-6f48-4b07-9504-8feddb48a882",
2021-07-15T20:55:47.3954357Z       "credits" : 0.5,
2021-07-15T20:55:47.3954785Z       "positive" : true,
2021-07-15T20:55:47.3955236Z       "type" : "MANUAL"
2021-07-15T20:55:47.3955621Z     }, {
2021-07-15T20:55:47.3956063Z       "id" : 2390,
2021-07-15T20:55:47.3957144Z       "reference" : "ActivityControlFlow:fa4fb017-4022-4b44-8caa-fe030173b8c5",
2021-07-15T20:55:47.3957925Z       "credits" : 0.5,
2021-07-15T20:55:47.3958352Z       "positive" : true,
2021-07-15T20:55:47.3958801Z       "type" : "MANUAL"
2021-07-15T20:55:47.3959395Z     }, {
2021-07-15T20:55:47.3959732Z       "id" : 2391,
2021-07-15T20:55:47.3960812Z       "reference" : "ActivityControlFlow:c479f73c-8125-40de-8e6b-0e67ea515307",
2021-07-15T20:55:47.3965929Z       "credits" : 0.5,
2021-07-15T20:55:47.3966386Z       "positive" : true,
2021-07-15T20:55:47.3966827Z       "type" : "MANUAL"
2021-07-15T20:55:47.3967218Z     }, {
2021-07-15T20:55:47.3967568Z       "id" : 2392,
2021-07-15T20:55:47.3968574Z       "reference" : "ActivityControlFlow:d268d8d8-8785-4f54-9777-4fa15e008728",
2021-07-15T20:55:47.3969302Z       "credits" : 0.5,
2021-07-15T20:55:47.3969747Z       "positive" : true,
2021-07-15T20:55:47.3970184Z       "type" : "MANUAL"
2021-07-15T20:55:47.3970577Z     }, {
2021-07-15T20:55:47.3970926Z       "id" : 2393,
2021-07-15T20:55:47.3972105Z       "reference" : "ActivityControlFlow:fa9b870d-2b4b-47e1-946a-abf365dc0f15",
2021-07-15T20:55:47.3997331Z       "credits" : 0.5,
2021-07-15T20:55:47.3997839Z       "positive" : true,
2021-07-15T20:55:47.3998322Z       "type" : "MANUAL"
2021-07-15T20:55:47.3998706Z     }, {
2021-07-15T20:55:47.3999060Z       "id" : 2394,
2021-07-15T20:55:47.4000244Z       "reference" : "ActivityControlFlow:84cb94c0-dc0a-4794-b8b1-83c44b3be13e",
2021-07-15T20:55:47.4001262Z       "credits" : 0.0,
2021-07-15T20:55:47.4001715Z       "positive" : true,
2021-07-15T20:55:47.4002176Z       "type" : "MANUAL"
2021-07-15T20:55:47.4002550Z     }, {
2021-07-15T20:55:47.4002904Z       "id" : 2395,
2021-07-15T20:55:47.4003830Z       "reference" : "ActivityControlFlow:98368d15-25a3-4572-a6b6-3fc97e0d532f",
2021-07-15T20:55:47.4004547Z       "credits" : 0.5,
2021-07-15T20:55:47.4004974Z       "positive" : true,
2021-07-15T20:55:47.4005424Z       "type" : "MANUAL"
2021-07-15T20:55:47.4005811Z     }, {
2021-07-15T20:55:47.4006144Z       "id" : 2396,
2021-07-15T20:55:47.4006582Z       "text" : "nope nope nope",
2021-07-15T20:55:47.4007644Z       "reference" : "ActivityControlFlow:7e59aded-6e55-4241-ac61-ab306ea23ecd",
2021-07-15T20:55:47.4008571Z       "credits" : -0.5,
2021-07-15T20:55:47.4009030Z       "positive" : false,
2021-07-15T20:55:47.4009487Z       "type" : "MANUAL"
2021-07-15T20:55:47.4009875Z     }, {
2021-07-15T20:55:47.4010207Z       "id" : 2397,
2021-07-15T20:55:47.4011145Z       "reference" : "ActivityControlFlow:9fb27fb7-153c-4938-92a7-3f79722cf5c0",
2021-07-15T20:55:47.4011893Z       "credits" : 0.5,
2021-07-15T20:55:47.4012320Z       "positive" : true,
2021-07-15T20:55:47.4012772Z       "type" : "MANUAL"
2021-07-15T20:55:47.4013156Z     }, {
2021-07-15T20:55:47.4013493Z       "id" : 2398,
2021-07-15T20:55:47.4013907Z       "text" : "not correct",
2021-07-15T20:55:47.4014922Z       "reference" : "ActivityControlFlow:a14b46df-64aa-401a-87f1-82076ba20e9c",
2021-07-15T20:55:47.4015828Z       "credits" : -0.5,
2021-07-15T20:55:47.4016285Z       "positive" : false,
2021-07-15T20:55:47.4016739Z       "type" : "MANUAL"
2021-07-15T20:55:47.4017127Z     }, {
2021-07-15T20:55:47.4017459Z       "id" : 2398,
2021-07-15T20:55:47.4017860Z       "text" : "correct",
2021-07-15T20:55:47.4018852Z       "reference" : "ActivityControlFlow:50a3baf7-7678-446a-ac5b-e491210ea673",
2021-07-15T20:55:47.4019601Z       "credits" : 0.5,
2021-07-15T20:55:47.4020043Z       "positive" : true,
2021-07-15T20:55:47.4020494Z       "type" : "MANUAL"
2021-07-15T20:55:47.4020867Z     } ]
2021-07-15T20:55:47.4021260Z         Session Attrs = {}
2021-07-15T20:55:47.4021560Z 
2021-07-15T20:55:47.4131667Z     Handler:
2021-07-15T20:55:47.4133066Z                  Type = de.tum.in.www1.artemis.web.rest.ModelingAssessmentResource
2021-07-15T20:55:47.4135403Z                Method = de.tum.in.www1.artemis.web.rest.ModelingAssessmentResource#saveModelingAssessment(long, long, boolean, List)
2021-07-15T20:55:47.4136779Z 
2021-07-15T20:55:47.4137135Z     Async:
2021-07-15T20:55:47.4137573Z         Async started = false
2021-07-15T20:55:47.4138045Z          Async result = null
2021-07-15T20:55:47.4138737Z 
2021-07-15T20:55:47.4139163Z     Resolved Exception:
2021-07-15T20:55:47.4140521Z                  Type = org.springframework.dao.InvalidDataAccessApiUsageException
2021-07-15T20:55:47.4142047Z 
2021-07-15T20:55:47.4142427Z     ModelAndView:
2021-07-15T20:55:47.4142862Z             View name = null
2021-07-15T20:55:47.4143548Z                  View = null
2021-07-15T20:55:47.4143983Z                 Model = null
2021-07-15T20:55:47.4144263Z 
2021-07-15T20:55:47.4144631Z     FlashMap:
2021-07-15T20:55:47.4145066Z            Attributes = null
2021-07-15T20:55:47.4145371Z 
2021-07-15T20:55:47.4145937Z     MockHttpServletResponse:
2021-07-15T20:55:47.4146510Z                Status = 500
2021-07-15T20:55:47.4146935Z         Error message = null
2021-07-15T20:55:47.4151870Z               Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Version:"1.3.3-beta7", Content-Type:"application/problem+json", X-Content-Type-Options:"nosniff", X-XSS-Protection:"1; mode=block", Cache-Control:"no-cache, no-store, max-age=0, must-revalidate", Pragma:"no-cache", Expires:"0", X-Frame-Options:"DENY", Content-Security-Policy:"script-src 'self' 'unsafe-inline' 'unsafe-eval'", Referrer-Policy:"strict-origin-when-cross-origin", Feature-Policy:"geolocation 'none'; midi 'none'; sync-xhr 'none'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; fullscreen *; payment 'none'"]
2021-07-15T20:55:47.4155761Z          Content type = application/problem+json
2021-07-15T20:55:47.4156319Z                  Body = {
2021-07-15T20:55:47.4157375Z       "type" : "https://www.jhipster.tech/problem/problem-with-message",
2021-07-15T20:55:47.4158261Z       "title" : "Internal Server Error",
2021-07-15T20:55:47.4158748Z       "status" : 500,
2021-07-15T20:55:47.4165913Z       "detail" : "Multiple representations of the same entity [de.tum.in.www1.artemis.domain.Feedback#2386] are being merged. Detached: [Feedback{id=2386, text='null', detailText='null', reference='ActivityActionNode:1c0576f6-89b0-4b13-96d9-e1a0a227e8e1', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]; Detached: [Feedback{id=2386, text='correct activity, but missing elements', detailText='null', reference='Activity:cfcb7b17-0500-4654-89bb-a7082d5c3b6d', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]; nested exception is java.lang.IllegalStateException: Multiple representations of the same entity [de.tum.in.www1.artemis.domain.Feedback#2386] are being merged. Detached: [Feedback{id=2386, text='null', detailText='null', reference='ActivityActionNode:1c0576f6-89b0-4b13-96d9-e1a0a227e8e1', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]; Detached: [Feedback{id=2386, text='correct activity, but missing elements', detailText='null', reference='Activity:cfcb7b17-0500-4654-89bb-a7082d5c3b6d', positive='true', type='MANUAL, visibility=null, gradingInstruction='null'}]",
2021-07-15T20:55:47.4172052Z       "path" : "/api/modeling-submissions/3579/result/0/assessment",
2021-07-15T20:55:47.4172794Z       "message" : "error.http.500"
2021-07-15T20:55:47.4173264Z     }
2021-07-15T20:55:47.4173677Z         Forwarded URL = null
2021-07-15T20:55:47.4174175Z        Redirected URL = null
2021-07-15T20:55:47.4174623Z               Cookies = []
2021-07-15T20:55:47.4901065Z 
2021-07-15T20:55:47.4903416Z ModelingAssessmentIntegrationTest > testAutomaticAssessmentActivityDiagramUponAssessmentSubmission() FAILED
2021-07-15T20:55:47.4905659Z     java.lang.AssertionError: Response status expected:<200> but was:<500>
2021-07-15T20:55:47.4907274Z         at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
2021-07-15T20:55:47.4909670Z         at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
2021-07-15T20:55:47.4912652Z         at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$is$1(StatusResultMatchers.java:63)
2021-07-15T20:55:47.4915241Z         at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:196)
2021-07-15T20:55:47.4917115Z         at de.tum.in.www1.artemis.util.RequestUtilService.put(RequestUtilService.java:333)
2021-07-15T20:55:47.4920351Z         at de.tum.in.www1.artemis.ModelingAssessmentIntegrationTest.createAssessment(ModelingAssessmentIntegrationTest.java:1389)
2021-07-15T20:55:47.4926671Z         at de.tum.in.www1.artemis.ModelingAssessmentIntegrationTest.testAutomaticAssessmentActivityDiagramUponAssessmentSubmission(ModelingAssessmentIntegrationTest.java:686)
```

--- a/src/test/resources/test-data/model-assessment/example-activity-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-activity-assessment.json
@@ -1,6 +1,5 @@
 [
     {
-        "id": 2378,
         "text": "correct activity, but missing elements",
         "reference": "Activity:cfcb7b17-0500-4654-89bb-a7082d5c3b6d",
         "credits": 0.0,
@@ -8,14 +7,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "ActivityControlFlow:fa0285e3-a3e6-4281-8c0e-223e5758106e",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "Activity:f85b009b-3320-4c73-9c7b-0d4bbe7e74f9",
         "credits": 0.5,
@@ -23,7 +20,6 @@
         "type": "MANUAL"
     },
     {
-        "id": 2381,
         "reference": "ActivityMergeNode:02c342c2-53d4-4bd7-9e08-774dbaa97ca6",
         "credits": 0.5,
         "positive": true,
@@ -37,14 +33,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2383,
         "reference": "ActivityInitialNode:2d1fbb07-bf49-43c1-a55f-35437e2cfcdc",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2384,
         "text": "not necessary, but fine",
         "reference": "ActivityMergeNode:e81a38a1-9ceb-46c9-b9e4-f8d274c76085",
         "credits": 0.0,
@@ -52,21 +46,18 @@
         "type": "MANUAL"
     },
     {
-        "id": 2385,
         "reference": "ActivityFinalNode:5019a714-3bef-4e4e-b649-b759c2b8778a",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2386,
         "reference": "ActivityActionNode:1c0576f6-89b0-4b13-96d9-e1a0a227e8e1",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2387,
         "text": "wrong name",
         "reference": "ActivityActionNode:056d0fe3-942a-4719-9b59-4c52443adcd5",
         "credits": -0.5,
@@ -74,7 +65,6 @@
         "type": "MANUAL"
     },
     {
-        "id": 2389,
         "text": "awesome",
         "reference": "ActivityObjectNode:525b1697-6f48-4b07-9504-8feddb48a882",
         "credits": 0.5,
@@ -82,49 +72,42 @@
         "type": "MANUAL"
     },
     {
-        "id": 2390,
         "reference": "ActivityControlFlow:fa4fb017-4022-4b44-8caa-fe030173b8c5",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2391,
         "reference": "ActivityControlFlow:c479f73c-8125-40de-8e6b-0e67ea515307",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2392,
         "reference": "ActivityControlFlow:d268d8d8-8785-4f54-9777-4fa15e008728",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2393,
         "reference": "ActivityControlFlow:fa9b870d-2b4b-47e1-946a-abf365dc0f15",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2394,
         "reference": "ActivityControlFlow:84cb94c0-dc0a-4794-b8b1-83c44b3be13e",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2395,
         "reference": "ActivityControlFlow:98368d15-25a3-4572-a6b6-3fc97e0d532f",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2396,
         "text": "nope nope nope",
         "reference": "ActivityControlFlow:7e59aded-6e55-4241-ac61-ab306ea23ecd",
         "credits": -0.5,
@@ -132,14 +115,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2397,
         "reference": "ActivityControlFlow:9fb27fb7-153c-4938-92a7-3f79722cf5c0",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2398,
         "text": "not correct",
         "reference": "ActivityControlFlow:a14b46df-64aa-401a-87f1-82076ba20e9c",
         "credits": -0.5,
@@ -147,7 +128,6 @@
         "type": "MANUAL"
     },
     {
-        "id": 2398,
         "text": "correct",
         "reference": "ActivityControlFlow:50a3baf7-7678-446a-ac5b-e491210ea673",
         "credits": 0.5,

--- a/src/test/resources/test-data/model-assessment/example-communication-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-communication-assessment.json
@@ -1,6 +1,5 @@
 [
     {
-        "id": 2378,
         "text": "correct activity, but missing elements",
         "reference": "ObjectName:31fa86c8-c245-4e30-aa47-328d1fd2d2c0",
         "credits": 0.0,
@@ -8,14 +7,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "ObjectAttribute:2bc1c0f7-a674-4cda-b496-c038d406aa69",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "ObjectMethod:1f4fb014-9fc6-4b57-a3d6-de23021600d1",
         "credits": 0.5,
@@ -23,28 +20,24 @@
         "type": "MANUAL"
     },
     {
-        "id": 2381,
         "reference": "ObjectName:3d3a94d1-7922-44d3-8914-12d11733b7ee",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2382,
         "reference": "ObjectAttribute:f79a4b0c-14aa-4c90-b1d7-a96e0adb02d2",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2383,
         "reference": "ObjectMethod:b7101509-7573-40ab-80ba-3c7ceceb401d",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2384,
         "text": "not necessary, but fine",
         "reference": "CommunicationLink:30e30816-dd49-4958-b9b0-87c9ee4a1eec",
         "credits": 0.0,

--- a/src/test/resources/test-data/model-assessment/example-component-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-component-assessment.json
@@ -1,6 +1,5 @@
 [
     {
-        "id": 2378,
         "text": "correct activity, but missing elements",
         "reference": "Component:02a071a3-9a29-4c31-b84b-a88c0c9e93cb",
         "credits": 0.0,
@@ -8,14 +7,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "ComponentInterface:2d277afa-ee74-4db1-ae43-8a4ee828c9f9",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "Component:6be596f0-cef1-45b8-8eec-fd6f1d910440",
         "credits": 0.5,
@@ -23,21 +20,18 @@
         "type": "MANUAL"
     },
     {
-        "id": 2381,
         "reference": "ComponentInterfaceProvided:e97fbb12-c23a-438e-96e4-37e55f0e5f44",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2382,
         "reference": "ComponentDependency:af93563d-d3f1-477a-b62a-af322327899b",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2383,
         "reference": "ComponentInterfaceRequired:d2ff0bff-5e46-415b-af2e-2a9f05dc6472",
         "credits": 0.0,
         "positive": true,

--- a/src/test/resources/test-data/model-assessment/example-deployment-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-deployment-assessment.json
@@ -1,6 +1,5 @@
 [
     {
-        "id": 2378,
         "text": "correct activity, but missing elements",
         "reference": "DeploymentNode:6e1de192-ce9e-4dfa-bb15-64afd964c232",
         "credits": 0.0,
@@ -8,14 +7,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "DeploymentComponent:a9f53b31-1041-4b78-84dc-e2c51d5cb153",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "DeploymentArtifact:a2366c55-c206-494d-aef8-f0672558ca0d",
         "credits": 0.5,
@@ -23,28 +20,24 @@
         "type": "MANUAL"
     },
     {
-        "id": 2381,
         "reference": "DeploymentInterface:d5e2bc8f-3851-44a0-927c-4f2caba9ac64",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2382,
         "reference": "DeploymentComponent:1bd706f5-5aa2-4945-967a-f7c6d5180a81",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2383,
         "reference": "DeploymentInterfaceProvided:2d215ebc-6574-46c6-b437-901572e8ad47",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2384,
         "text": "not necessary, but fine",
         "reference": "DeploymentAssociation:e51e25e8-2767-49ed-aec1-5bea9232c213",
         "credits": 0.0,
@@ -52,14 +45,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2385,
         "reference": "DeploymentDependency:059cd7ba-a2fd-49b4-b9ab-68574dbc9d5c",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2386,
         "reference": "DeploymentInterfaceRequired:0ac07a9d-fd5e-4920-ae15-a509e586de65",
         "credits": 0.5,
         "positive": true,

--- a/src/test/resources/test-data/model-assessment/example-flowchart-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-flowchart-assessment.json
@@ -8,14 +8,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "FlowchartProcess:f47952c4-8760-400a-a8ea-d18f9c327cdd",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "FlowchartDecision:d57161a9-7d4b-4673-a1fb-1529cb393922",
         "credits": 0.5,
@@ -23,28 +21,24 @@
         "type": "MANUAL"
     },
     {
-        "id": 2381,
         "reference": "FlowchartInputOutput:e1dfee84-b78a-48f6-8db7-1051da8607cb",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2382,
         "reference": "FlowchartFunctionCall:50280dd9-15ee-4477-89d7-1d3ade2ccd2e",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2383,
         "reference": "FlowchartFlowline:d86bf9cf-2f19-4c0d-9879-503f43e09bce",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2384,
         "text": "not necessary, but fine",
         "reference": "FlowchartFlowline:925c0d32-e46d-43f6-b6d0-713c0fd9231e",
         "credits": 0.0,
@@ -52,14 +46,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2385,
         "reference": "FlowchartFlowline:49e82320-7084-4024-838f-ec5552bc4389",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2386,
         "reference": "FlowchartFlowline:ef29eef1-1ffe-4b49-9b21-d6a9ea19e2d1",
         "credits": 0.5,
         "positive": true,

--- a/src/test/resources/test-data/model-assessment/example-object-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-object-assessment.json
@@ -1,6 +1,5 @@
 [
     {
-        "id": 2378,
         "text": "correct activity, but missing elements",
         "reference": "ObjectName:0129ed3d-efe1-4d46-a9ae-7de56273571e",
         "credits": 0.0,
@@ -8,14 +7,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "ObjectAttribute:f716c6e6-b4ba-49e3-94b7-fe634d9dbb38",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "ObjectMethod:84241ba6-4366-4cff-bab7-a73deb0e8013",
         "credits": 0.5,
@@ -23,28 +20,24 @@
         "type": "MANUAL"
     },
     {
-        "id": 2381,
         "reference": "ObjectName:278bd318-7bf8-405a-96c4-05aa8a43de0f",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2382,
         "reference": "ObjectAttribute:18088b91-2b8a-44a7-b62a-cad59c0955b2",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2383,
         "reference": "ObjectMethod:ca37589d-d03a-4ca0-bfb3-1472eedd9dab",
         "credits": 0.0,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2384,
         "text": "not necessary, but fine",
         "reference": "ObjectLink:84dfdc22-dd73-403f-84df-141125f3096a",
         "credits": 0.0,

--- a/src/test/resources/test-data/model-assessment/example-petri-net-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-petri-net-assessment.json
@@ -1,6 +1,5 @@
 [
     {
-        "id": 2378,
         "text": "correct activity, but missing elements",
         "reference": "PetriNetTransition:b5120d7d-24fe-465c-880e-cf35cfb75d34",
         "credits": 0.0,
@@ -8,14 +7,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "PetriNetPlace:b62dff7b-0c20-4573-bc1c-38c94fedffd4",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "PetriNetArc:ae4c0231-9a8a-4d90-b8b4-a5332769eb14",
         "credits": 0.5,

--- a/src/test/resources/test-data/model-assessment/example-syntax-tree-assessment.json
+++ b/src/test/resources/test-data/model-assessment/example-syntax-tree-assessment.json
@@ -1,6 +1,5 @@
 [
     {
-        "id": 2378,
         "text": "correct activity, but missing elements",
         "reference": "SyntaxTreeNonterminal:99c3538a-cc7c-4d79-9eec-8368aa7be6dc",
         "credits": 0.0,
@@ -8,14 +7,12 @@
         "type": "MANUAL"
     },
     {
-        "id": 2379,
         "reference": "SyntaxTreeTerminal:c441d34a-2202-458c-b8b0-3ac7c67ae77f",
         "credits": 0.5,
         "positive": true,
         "type": "MANUAL"
     },
     {
-        "id": 2380,
         "text": "good",
         "reference": "SyntaxTreeLink:0756d15f-c038-4224-b278-3997554edab4",
         "credits": 0.5,

--- a/src/test/resources/test-data/model-assessment/object-assessment.json
+++ b/src/test/resources/test-data/model-assessment/object-assessment.json
@@ -1,6 +1,5 @@
 [
   {
-    "id": 600,
     "text": "nice",
     "reference": "ObjectName:79215533-b25b-4f39-8619-f894c7f72e2e",
     "credits": 1.5,
@@ -10,7 +9,6 @@
     "referenceId": "79215533-b25b-4f39-8619-f894c7f72e2e"
   },
   {
-    "id": 601,
     "text": "good",
     "reference": "ObjectAttribute:f0f62a6e-3997-4837-90b2-5bc5408e886d",
     "credits": 0.5,
@@ -20,7 +18,6 @@
     "referenceId": "f0f62a6e-3997-4837-90b2-5bc5408e886d"
   },
   {
-    "id": 602,
     "text": "whoop whoop",
     "reference": "ObjectLink:86a0210a-5d84-41cf-b4d1-9cc2a5408df4",
     "credits": 1,
@@ -30,7 +27,6 @@
     "referenceId": "86a0210a-5d84-41cf-b4d1-9cc2a5408df4"
   },
   {
-    "id": 603,
     "text": "this is not correct",
     "reference": "ObjectAttribute:3290730a-e825-4561-81d7-735e0034dc87",
     "credits": -0.5,
@@ -40,7 +36,6 @@
     "referenceId": "3290730a-e825-4561-81d7-735e0034dc87"
   },
   {
-    "id": 604,
     "text": "correct",
     "reference": "ObjectName:70fc28ff-6d2b-46f8-aa90-ec7f7a950ec0",
     "credits": 2,

--- a/src/test/resources/test-data/model-assessment/use-case-assessment.json
+++ b/src/test/resources/test-data/model-assessment/use-case-assessment.json
@@ -1,6 +1,5 @@
 [
   {
-    "id": 605,
     "text": "good use case",
     "reference": "UseCase:cd561d6c-022d-43ed-b023-f8485413a702",
     "credits": 1,
@@ -10,7 +9,6 @@
     "referenceId": "cd561d6c-022d-43ed-b023-f8485413a702"
   },
   {
-    "id": 606,
     "text": "correct",
     "reference": "UseCase:6c07c681-1123-4e65-8dc9-32992aab5cb3",
     "credits": 1,
@@ -20,7 +18,6 @@
     "referenceId": "6c07c681-1123-4e65-8dc9-32992aab5cb3"
   },
   {
-    "id": 607,
     "text": "association to wrong use case",
     "reference": "UseCaseAssociation:aa3e5ec0-a185-4834-8c31-2dfa85c5a1ca",
     "credits": -0.5,
@@ -30,7 +27,6 @@
     "referenceId": "aa3e5ec0-a185-4834-8c31-2dfa85c5a1ca"
   },
   {
-    "id": 608,
     "reference": "UseCaseActor:ca8fb56a-b0a4-436f-ab2f-03ae8d267165",
     "credits": 1,
     "positive": true,
@@ -39,7 +35,6 @@
     "referenceId": "ca8fb56a-b0a4-436f-ab2f-03ae8d267165"
   },
   {
-    "id": 609,
     "reference": "UseCaseActor:2a2d0d80-06da-4a2e-918a-4ef202b7e765",
     "credits": 1,
     "positive": true,
@@ -48,7 +43,6 @@
     "referenceId": "2a2d0d80-06da-4a2e-918a-4ef202b7e765"
   },
   {
-    "id": 610,
     "text": "good",
     "reference": "UseCaseAssociation:5a2e91fc-b2cd-4121-89a4-196a06e5eac6",
     "credits": 0.5,
@@ -58,7 +52,6 @@
     "referenceId": "5a2e91fc-b2cd-4121-89a4-196a06e5eac6"
   },
   {
-    "id": 611,
     "text": "correct",
     "reference": "UseCaseAssociation:65dd6377-ec73-441b-8174-0427ab677662",
     "credits": 0.5,
@@ -68,7 +61,6 @@
     "referenceId": "65dd6377-ec73-441b-8174-0427ab677662"
   },
   {
-    "id": 612,
     "reference": "UseCaseSystem:caa18e16-04d3-4de1-938c-db21ce4ae368",
     "credits": 1.5,
     "positive": true,

